### ISSUE TITLE
Cannot find the integrated GPU. Is this an Optimus system

### ISSUE THREAD

**Issue by thamudi** [Askannz/optimus-manager#486]
The issue is that I cannot get optimus manager to start properly after setup  
I have both AMD ryzen 7 and AMD Radeon Graphics with Nvidia RTX 3050.

**System info**
- **Distribution**: Manjaro, Linux _5.15_ LTS 
- **Desktop Environment**: KDE _5.24_
- **Display Manager**: SDDM. I did follow up the pre-install instructions and commented out the `DisplayCommand` and the `DisplayStopCommand`
- **Laptop Device**: Legion 5 _15ACH6_ 
- **Optimus-manager** version: Latest stable release (optimus-manager)

**Logs**
The error I get when I run `optimus-manager` is the following
```
Cannot find the integrated GPU. Is this an Optimus system
```

I took a look around the forum and some people said that they switched over to AMD support branch. I took a look around in the build files (I am installing it from the AUR) and there was none mentioning which branch to switch to.

Any suggestions?

**Comment by thamudi** [comment]
**Update**

As mentioned in the other Issue, after a fresh install I get this issue:

`Command 'modprobe -r nvidia_drm nvidia_modeset nvidia_uvm nvidia' returned non-zero exit status 1.`

```bash
[8] INFO: # Xorg pre-start hook
[8] INFO: Previous state was: {'type': 'pending_pre_xorg_start', 'requested_mode': 'integrated', 'current_mode': None}
[8] INFO: Requested mode is: integrated
[709] INFO: Available modules: ['nouveau', 'nvidia', 'nvidia_drm', 'nvidia_modeset', 'nvidia_uvm']
[709] INFO: Unloading modules ['nvidia_drm', 'nvidia_modeset', 'nvidia_uvm', 'nvidia'] (if loaded)
[713] INFO: Cannot unload modules: None
[713] INFO: Waiting 1s and retrying.
[1719] INFO: Cannot unload modules: None
[1719] INFO: Waiting 1s and retrying.
[2726] INFO: Cannot unload modules: None
[2726] INFO: Waiting 1s and retrying.
[3730] INFO: Cannot unload modules: None
[3731] INFO: Waiting 1s and retrying.
[4737] INFO: Cannot unload modules: None
[4737] INFO: Waiting 1s and retrying.
[5746] INFO: Max tries (6) exceeded
[5747] ERROR: Xorg pre-start setup error
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/optimus_manager/kernel.py", line 269, in _unload_modules
    subprocess.check_call(
  File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'modprobe -r nvidia_drm nvidia_modeset nvidia_uvm nvidia' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/optimus_manager/hooks/pre_xorg_start.py", line 51, in main
    setup_kernel_state(config, prev_state, requested_mode)
  File "/usr/lib/python3.10/site-packages/optimus_manager/kernel.py", line 25, in setup_kernel_state
    _nvidia_down(config)
  File "/usr/lib/python3.10/site-packages/optimus_manager/kernel.py", line 104, in _nvidia_down
    _unload_nvidia_modules(available_modules)
  File "/usr/lib/python3.10/site-packages/optimus_manager/kernel.py", line 214, in _unload_nvidia_modules
    _unload_modules(available_modules, ["nvidia_drm", "nvidia_modeset", "nvidia_uvm", "nvidia"])
  File "/usr/lib/python3.10/site-packages/optimus_manager/kernel.py", line 277, in _unload_modules
    raise KernelSetupError(f"Cannot unload modules {modules_to_unload}: {e.stderr}") from e
optimus_manager.kernel.KernelSetupError: Cannot unload modules ['nvidia_drm', 'nvidia_modeset', 'nvidia_uvm', 'nvidia']: None
[5749] INFO: Removing /etc/X11/xorg.conf.d/10-optimus-manager.conf (if present)
[5749] INFO: Writing state {'type': 'pre_xorg_start_failed', 'switch_id': '20220324T091104', 'requested_mode': 'integrated'}
```

**Comment by thamudi** [comment]
**Update/ Edit:**
I regret posting the previous comment so quickly. I was finally able to get it to work.
So Here is how I was able to fix the situation!

First, I uninstalled system76 power manager for good.

Second, I switched form Discrete graphics to **Switchable graphics**. This was done in the BIOS. Note: I am using Legion 5.

Third, I went and uninstalled the video-nvidia driver for the display controller. This is what was causing my issue. When I installed Manjaro on it, it had the discrete graphics on from the bios and installed only the nvidia graphics driver. This is why I wasn't able to unload and switch, explained how fix in step 2.

Fourth, I installed the video-hybrid-amd-nvidia-prime none free and restarted my computer after that to make sure everything is working fine.

Lastly, I installed optimus manager and optimus manager-qt, enable/ start the service and done a last restart after that because i love restarting my system and everything was working like a charm.

I faced some issue with brightness, I was able to fix it by editing some params in the grub. Here's the [resource](https://forum.manjaro.org/t/amdgpu-cant-adjust-brightness-in-hybrid-mode-with-nvidia-discrete-works/90729/3) to where I was able to fix that issue.

Hope this helps somebody else out there.
Cheers! 👋🏼
